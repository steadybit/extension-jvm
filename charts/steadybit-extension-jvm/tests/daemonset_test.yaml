templates:
  - daemonset.yaml
chart:
  appVersion: v0.0.0
capabilities:
  majorVersion: 1
  minorVersion: 29
tests:
  - it: manifest should match snapshot with appArmorProfile for k8s >= 1.30
    capabilities:
      majorVersion: 1
      minorVersion: 30
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot using docker
    set:
      container:
        runtime: docker
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot using containerd and resources
    set:
      container:
        runtime: containerd
    resources:
      limits:
        cpu: 100m
        memory: 128Mi
      requests:
        cpu: 200m
        memory: 256Mi
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot using crio using podAnnotations and Labels
    set:
      container:
        runtime: cri-o
      podAnnotations:
        some-annotation: "some-annotation-value"
      podLabels:
        some-label: "some-label-value"
    asserts:
      - matchSnapshot: { }
  - it: manifest should fail without runtime
    set:
      container:
        runtime: null
    asserts:
      - failedTemplate:
          errorMessage: "unknown container runtime: <nil> (must be one of containerd, cri-o, docker)"
  - it: manifest should match snapshot with TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with mutual TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
        client:
          certificates:
            fromSecrets:
              - client-cert-a
              - client-cert-a
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with mutual TLS using containerPaths
    set:
      tls:
        server:
          certificate:
            path: /etc/tls/server.crt
            key:
              path: /etc/tls/server.key
        client:
          certificates:
            paths:
              - /etc/tls/ca.crt
              - /etc/tls/ca2.crt
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with extra env vars
    set:
      extraEnv:
        - name: FOO
          value: "bar"
      extraEnvFrom:
        - configMapRef:
          name: env-configmap
        - secretRef:
          name: env-secrets
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with different containerPorts
    set:
      containerPorts:
        http: 9090
        health: 9091
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with extra labels
    set:
      extraLabels:
        tags.datadoghq.com/service: steadybit-extension
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with update strategy
    set:
      updateStrategy:
        type: RollingUpdate
        rollingUpdate:
          maxUnavailable: 5
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with podSecurityContext
    set:
      podSecurityContext:
        runAsUser: 2222
    asserts:
      - matchSnapshot: {}

  - it: manifest should match snapshot with priority class
    set:
      priorityClassName: my-priority-class
    asserts:
      - matchSnapshot: {}
