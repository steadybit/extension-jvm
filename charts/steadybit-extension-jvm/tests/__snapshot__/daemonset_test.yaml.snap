manifest should match snapshot with TLS:
  1: |
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-jvm
      namespace: NAMESPACE
    spec:
      selector:
        matchLabels:
          app: steadybit-extension-jvm
          app.kubernetes.io/name: steadybit-extension-jvm
      template:
        metadata:
          annotations:
            container.apparmor.security.beta.kubernetes.io/steadybit-extension-jvm: unconfined
            steadybit.com/extension-auto-discovery: "{\n  \"extensions\": [\n    {\n      \"port\": 8085,\n      \"types\": [\"ACTION\",\"DISCOVERY\"],\n      \"tls\": {\n        \"server\": {\n          \"extraCertsFile\": \"server-cert/tls.crt\"\n        }\n        \n      }\n    }\n  ]\n}\n"
          labels:
            app: steadybit-extension-jvm
            app.kubernetes.io/name: steadybit-extension-jvm
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
              image: ghcr.io/steadybit/extension-jvm:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8081
              name: steadybit-extension-jvm
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8081
              securityContext:
                capabilities:
                  add:
                    - SYS_ADMIN
                    - SYS_RESOURCE
                    - SYS_PTRACE
                    - KILL
                    - NET_ADMIN
                    - DAC_OVERRIDE
                    - SETUID
                    - SETGID
                    - AUDIT_WRITE
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
                seccompProfile:
                  type: Unconfined
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          dnsPolicy: ClusterFirstWithHostNet
          hostNetwork: true
          hostPID: true
          volumes:
            - emptyDir: {}
              name: tmp-dir
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with extra env vars:
  1: |
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-jvm
      namespace: NAMESPACE
    spec:
      selector:
        matchLabels:
          app: steadybit-extension-jvm
          app.kubernetes.io/name: steadybit-extension-jvm
      template:
        metadata:
          annotations:
            container.apparmor.security.beta.kubernetes.io/steadybit-extension-jvm: unconfined
            steadybit.com/extension-auto-discovery: "{\n  \"extensions\": [\n    {\n      \"port\": 8085,\n      \"types\": [\"ACTION\",\"DISCOVERY\"],\n      \"tls\": {\n        \n      }\n    }\n  ]\n}\n"
          labels:
            app: steadybit-extension-jvm
            app.kubernetes.io/name: steadybit-extension-jvm
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: FOO
                  value: bar
              envFrom:
                - configMapRef: null
                  name: env-configmap
                - name: env-secrets
                  secretRef: null
              image: ghcr.io/steadybit/extension-jvm:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8081
              name: steadybit-extension-jvm
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8081
              securityContext:
                capabilities:
                  add:
                    - SYS_ADMIN
                    - SYS_RESOURCE
                    - SYS_PTRACE
                    - KILL
                    - NET_ADMIN
                    - DAC_OVERRIDE
                    - SETUID
                    - SETGID
                    - AUDIT_WRITE
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
                seccompProfile:
                  type: Unconfined
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          dnsPolicy: ClusterFirstWithHostNet
          hostNetwork: true
          hostPID: true
          volumes:
            - emptyDir: {}
              name: tmp-dir
manifest should match snapshot with mutual TLS:
  1: |
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-jvm
      namespace: NAMESPACE
    spec:
      selector:
        matchLabels:
          app: steadybit-extension-jvm
          app.kubernetes.io/name: steadybit-extension-jvm
      template:
        metadata:
          annotations:
            container.apparmor.security.beta.kubernetes.io/steadybit-extension-jvm: unconfined
            steadybit.com/extension-auto-discovery: |
              {
                "extensions": [
                  {
                    "port": 8085,
                    "types": ["ACTION","DISCOVERY"],
                    "tls": {
                      "server": {
                        "extraCertsFile": "server-cert/tls.crt"
                      },
                      "client": {
                        "certChainFile": "client-cert-a/tls.crt",
                        "certKeyFile": "client-cert-a/tls.key"
                      }
                    }
                  }
                ]
              }
          labels:
            app: steadybit-extension-jvm
            app.kubernetes.io/name: steadybit-extension-jvm
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/extension/certificates/client-cert-a/tls.crt,/etc/extension/certificates/client-cert-a/tls.crt
              image: ghcr.io/steadybit/extension-jvm:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8081
              name: steadybit-extension-jvm
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8081
              securityContext:
                capabilities:
                  add:
                    - SYS_ADMIN
                    - SYS_RESOURCE
                    - SYS_PTRACE
                    - KILL
                    - NET_ADMIN
                    - DAC_OVERRIDE
                    - SETUID
                    - SETGID
                    - AUDIT_WRITE
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
                seccompProfile:
                  type: Unconfined
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
                - mountPath: /etc/extension/certificates/client-cert-a
                  name: certificate-client-cert-a
                  readOnly: true
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          dnsPolicy: ClusterFirstWithHostNet
          hostNetwork: true
          hostPID: true
          volumes:
            - emptyDir: {}
              name: tmp-dir
            - name: certificate-client-cert-a
              secret:
                optional: false
                secretName: client-cert-a
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot without TLS:
  1: |
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-jvm
      namespace: NAMESPACE
    spec:
      selector:
        matchLabels:
          app: steadybit-extension-jvm
          app.kubernetes.io/name: steadybit-extension-jvm
      template:
        metadata:
          annotations:
            container.apparmor.security.beta.kubernetes.io/steadybit-extension-jvm: unconfined
            steadybit.com/extension-auto-discovery: "{\n  \"extensions\": [\n    {\n      \"port\": 8085,\n      \"types\": [\"ACTION\",\"DISCOVERY\"],\n      \"tls\": {\n        \n      }\n    }\n  ]\n}\n"
          labels:
            app: steadybit-extension-jvm
            app.kubernetes.io/name: steadybit-extension-jvm
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
              image: ghcr.io/steadybit/extension-jvm:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8081
              name: steadybit-extension-jvm
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8081
              securityContext:
                capabilities:
                  add:
                    - SYS_ADMIN
                    - SYS_RESOURCE
                    - SYS_PTRACE
                    - KILL
                    - NET_ADMIN
                    - DAC_OVERRIDE
                    - SETUID
                    - SETGID
                    - AUDIT_WRITE
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
                seccompProfile:
                  type: Unconfined
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          dnsPolicy: ClusterFirstWithHostNet
          hostNetwork: true
          hostPID: true
          volumes:
            - emptyDir: {}
              name: tmp-dir
